tests/variable_creation_test.py

```diff
@@ -475,6 +475,35 @@ def test_arange():
     assert sc.identical(var, expected)
 
 
+def test_arange_datetime_from_int():
+    var = sc.arange('t', 14, 65, 21, unit='s', dtype='datetime64')
+    expected = sc.datetimes(dims=['t'], values=[14, 35, 56], unit='s')
+    assert sc.identical(var, expected)
+
+
+def test_arange_datetime_from_np_datetime64():
+    var = sc.arange('t', np.datetime64('2022-08-02T11:18'),
+                    np.datetime64('2022-08-02T11:52'), 15)
+    expected = sc.datetimes(
+        dims=['t'],
+        values=['2022-08-02T11:18', '2022-08-02T11:33', '2022-08-02T11:48'],
+        unit='m')
```

Comment:
`m`? Or `min`? Why is this working?

Reply:
Oh, right! Not sure why this works.

Reply:
Probably because of this https://github.com/scipp/scipp/blob/fe34d19300029627c170fd58774168aa8c69308e/lib/python/dtype.cpp#L278

Reply:
Fixed